### PR TITLE
frontend/banner: add banner support for bitbox02

### PR DESCRIPTION
**Pull request by Beerosagos**
Banner messages are available for bitbox01 trough a dedicated backend endpoint that allows to fetch a banner json exposed on a specific url.

This adds banner support also for bitbox02, which allows e.g. to advertise offers to the users or communicate important messages inside the app.

### CODE REVIEW

**Review by thisconnect — APPROVED** [review]
untested LGTM

File: frontends/web/src/components/banner/banner.tsx [hunk 1]
```diff
@@ -24,8 +24,9 @@ type TBannerInfo = {
     message: { [key: string]: string; };
     link?: {
         href: string;
-        text: string;
+        text?: string;
```
[the remark this answers is not in the document]
Comment (Beerosagos) [reply frontends/web/src/components/banner/banner.tsx:27]: Yeah that surprised me too! Not used, and it was missing in the backend struct!

File: backend/banners/banners.go [hunk 2]
```diff
@@ -43,8 +44,12 @@ type Message struct {
 	ID string `json:"id"`
 	// Link, if present, will be appended to the message.
 	Link *struct {
-		Href string `json:"href"`
+		Href string  `json:"href"`
+		Text *string `json:"text"`
 	} `json:"link"`
+	// Dismissible: if true the ID field will be the key of config.frontend map to keep
+	// trace of dismissed banners (see 	status.tsx for details).
```
Comment (benma) [inline backend/banners/banners.go:51]: there is a tab after `see` :)

File: frontends/web/src/components/banner/banner.tsx [hunk 3]
```diff
@@ -44,11 +45,11 @@ function Banner({ banner, i18n, t }: TProps) {
     return null;
   }
   return banner && (
-    <Status dismissible="" type="warning">
+    <Status dismissible={banner.dismissible ? banner.id : ''} type="warning">
```
Comment (benma) [inline frontends/web/src/components/banner/banner.tsx:48]: so banner.id was unused?

**Review by benma — APPROVED** [review]

File: backend/banners/banners.go (hunk 2 above)
Comment (Beerosagos) [reply backend/banners/banners.go]: sry, my vim config has still some issues with tabs and spaces :sweat_smile: fixed

File: frontends/web/src/components/banner/banner.tsx (hunk 3 above)
Comment (Beerosagos) [reply frontends/web/src/components/banner/banner.tsx:48]: yeah, I think so

Comment (benma) [reply frontends/web/src/components/banner/banner.tsx:48]: it was used, see here:

https://github.com/digitalbitbox/bitbox-wallet-app/commit/a0601e2cb4330b8047ca15ff95597b1c22a69d9c#diff-694b6e6a9809368b1c603dba8dc008244ccb8e76febfb19ab760258dbe329ab8R47

makes sense to use the key as it used to be like this:

`dismissible={`banner-${msgKey}-${banner.id}`}` so the ID is scoped.

Comment (Beerosagos) [reply frontends/web/src/components/banner/banner.tsx]: Good point, thanks! Done

**Comment by Beerosagos** [comment]
Pushed a second commit to add support for `type` field, which allows different graphics for alert, info, and warning banners. @thisconnect PTAL :pray:

**Review by thisconnect — APPROVED** [review]
LGTM

**Pull request merged**